{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "RetryInvocationHandler.java",
  "functionName": "getFailException",
  "functionId": "getFailException",
  "sourceFilePath": "hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java",
  "functionStartLine": 264,
  "functionEndLine": 266,
  "numCommitsSeen": 39,
  "timeTaken": 665,
  "changeHistory": [
    "55796a0946f80a35055701a34379e374399009c5"
  ],
  "changeHistoryShort": {
    "55796a0946f80a35055701a34379e374399009c5": "Yintroduced"
  },
  "changeHistoryDetails": {
    "55796a0946f80a35055701a34379e374399009c5": {
      "type": "Yintroduced",
      "commitMessage": "HDFS-11395. RequestHedgingProxyProvider#RequestHedgingInvocationHandler hides the Exception thrown from NameNode. Contributed by Nandakumar.\n",
      "commitDate": "13/03/17 2:24 PM",
      "commitName": "55796a0946f80a35055701a34379e374399009c5",
      "commitAuthor": "Jing Zhao",
      "diff": "@@ -0,0 +1,3 @@\n+    Exception getFailException() {\n+      return failException;\n+    }\n\\ No newline at end of file\n",
      "actualSource": "    Exception getFailException() {\n      return failException;\n    }",
      "path": "hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java"
    }
  }
}